hyperspy/_components/pes_voigt.py

```diff
@@ -0,0 +1,331 @@
+
+    @property
+    def FWHM(self):
+        if legacy is False:
+            return super().sigma.value * sigma2fwhm
+
+    @FWHM.setter
+    def FWHM(self, value):
+        if legacy is False:
+            super(Voigt, self.__class__).sigma.value.fset(self, value
+                                                                / sigma2fwhm)
+
+    @property
+    def lwidth(self):
+        if legacy is False:
+            return super().gamma.value * 2
+
+    @lwidth.setter
+    def lwidth(self, value):
+        if legacy is False:
+            super(Voigt, self.__class__).gamma.value.fset(self, value / 2)
+
+
+class PESVoigt(Component):
+
+    """ Voigt component for photoemission spectroscopy data analysis.
+
+    Voigt profile component with support for shirley background,
+    non_isochromaticity, transmission_function corrections and spin orbit
+    splitting specially suited for photoemission spectroscopy data analysis.
+
+    .. math:: 
+        f(x) = G(x) \cdot L(x) 
+
+    where :math:`G(x)` is the Gaussian function and :math:`L(x)` is the 
+    Lorentzian function. This component uses an approximate formula by David 
+    (see Notes).
+
+
+    Parameters
+    ----------
+
+    area : Parameter
+        Intensity below the peak.
+    centre: Parameter
+        Location of the maximum of the peak.
+    FWHM : Parameter
+        FWHM=:math:`2 \sigma \sqrt(2 \log(2))` of the Gaussian distribution.
+    gamma : Parameter
+        :math:`\gamma` of the Lorentzian distribution.
+    resolution : Parameter
+    shirley_background : Parameter
+    non_isochromaticity : Parameter
+    transmission_function : Parameter
+    spin_orbit_splitting : Bool
+    spin_orbit_branching_ratio : float
+    spin_orbit_splitting_energy : float
+
+    Notes
+    -----
+    Uses an approximate formula according to
+    W.I.F. David, J. Appl. Cryst. (1986). 19, 63-64.
+    doi:10.1107/S0021889886089999
+    """
+
+    def __init__(self):
+        Component.__init__(self, (
+            'area',
+            'centre',
+            'FWHM',
+            'gamma',
+            'resolution',
+            'shirley_background',
+            'non_isochromaticity',
+            'transmission_function'))
+        self._position = self.centre
+        self.FWHM.value = 1
+        self.gamma.value = 0
+        self.area.value = 1
+        self.resolution.value = 0
+        self.resolution.free = False
+        self.shirley_background.free = False
+        self.non_isochromaticity.value = 0
+        self.non_isochromaticity.free = False
+        self.transmission_function.value = 1
+        self.transmission_function.free = False
+        # Options
+        self.shirley_background.active = False
+        self.spin_orbit_splitting = False
+        self.spin_orbit_branching_ratio = 0.5
+        self.spin_orbit_splitting_energy = 0.61
+        self.isbackground = False
+        self.convolved = True
+
+    def function(self, x):
```

Comment:
Doesn't this need a docstring?

Reply:
Same as above, since this is simply moving code around, this can be optional for PR.